src/common/tx-builder-base.ts

```diff
@@ -249,26 +253,42 @@ export class TxBuilderBase {
             },
         });
 
-        if (!result.confirmedLocalExecution) {
-            while (true) {
-                try {
-                    result = await this.client.getTransactionBlock({
-                        digest: result.digest,
-                        options: {
-                            showEffects: true,
-                            showObjectChanges: true,
-                            ...options,
-                        },
-                    });
-                    break;
-                } catch (e) {
-                    console.log(e);
-                    await new Promise((resolve) => setTimeout(resolve, 1000));
-                }
+        const maxRetries = 10; // 10 seconds with 1-second delay
+        let retries = 0;
+
+        while (true) {
```

Comment:
```suggestion
        while (retries < maxRetries) {
```
check max retry in the loop and throw error at the end of the function instead?